### PR TITLE
Always show chain native tokens in incoming funds table

### LINKED ISSUE #2787: Ether is disappearing from the Incoming funds list
## Steps to reproduce

1. Pay funds to a colony with the chains native token, i.e. Ether with contract address `0x0000000000000000000000000000000000000000`. (You can do this by adding a local wallet address to metamask on Ganache network.
2. Go to the Incoming funds page and accept the Ether incoming funds.
3. The token Ether will disappear from the list.
4. No others will dissappear.

## Expected behaviour

* You should still see the chain's native token in the Incoming funds list.

## Actual behaviour

* The chain's native token disappears from the Incoming funds list.

ether-dissappearing

### PR DESCRIPTION
**Pull request by davecreaser**
## Description

There was an [issue raised](https://github.com/JoinColony/colonyCDapp/issues/2787) where the chain's native token was disappearing from the Incoming Funds table after you claimed the funds. With the other tokens, the claim remains in a 'claimed' state. The problem is that the way chain native tokens are handled is very different - see the comments in the issue for more details on that.

This is a solution which will ensure there is no confusion when the chain native token is claimed and disappears from the list. In the future when financial reporting is worked on then this behaviour can be changed properly.

Here's a description of what this PR will do:

- Ensure that the chain's native token item will always be visible at the bottom of the Incoming Funds table.

Collapsed view:
- It will show a value of 0 if there is nothing to claim.
- If there are incoming funds, then it will show the value of funds to claim.

Expanded view:
- There will be an info row that always appears at the bottom of the expanded table for this token, regardless if there are funds to claim or not.
- Claimable incoming funds will appear above the notice and work as expected, allowing claiming with the claimed funds rows disappearing, keeping the note in place.

## Testing

You'll need to test that when you send ethereum to a colony, it shows up as expected. That includes before sending, after sending, and after claiming.

* Step 1. On a clean environment, go to the incoming funds page. You should see Ether at the bottom of the table with 0 Eth incoming. There should also be a warning message when you expand this row informing you that a history of claims won't be saved:

Screenshot 2024-08-21 at 12 07 30

* Step 2. Get the colony address. I like to do this just by going to the balances page and clicking on the `Add funds to the colony` button in the top right:

Screenshot 2024-08-21 at 12 05 31

* Step 3. Open a dev wallet in metamask, and send some ethereum to the colony. Ensure that you're on the correct network (Ganache Local) and that you're on the right account (dev wallet). If you need help setting up the network or importing a dev wallet let me know and I can help you do so. Also if this transaction fails, you may need to reset your nonces, again, just let me know and I can help you.

Screenshot 2024-08-21 at 12 09 09

* Step 4. Go back to the incoming funds page and refresh. You should now see some Eth available to be claimed.

Screenshot 2024-08-21 at 12 11 08

* Step 5. Accept the Eth, and you should see it be accepted (the button will disable while this is in progress) and then go back to zero. The row should _not_ disappear from the table.

Screenshot 2024-08-21 at 12 12 37

## Diffs

**Changes** 🏗

* Always return a claim from the `fetchColonyNativeFundsClaim` lambda, even if the amount is zero.
* Always show Eth at the bottom of the incoming funds table.
* Add a warning message to the incoming funds table.

Resolves #2787

### CODE REVIEW

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Great work on this Dave, everything is working nicely!

New claims show up in the table:

Screenshot 2024-08-22 at 10 01 51

Also sanity checked they get claimed properly:

Screenshot 2024-08-22 at 10 02 34

I noticed when I sent two payments to the colony, they were both summed together and showed as a single claim - this looks like the intended behaviour from the code, but just wanted to check it is expected?

The only issue I noticed is this icon shrinking once you get to small screen widths on mobile:

Screenshot 2024-08-22 at 10 06 25

Nice job!

File: amplify/backend/function/fetchColonyNativeFundsClaim/src/index.js [hunk 1]
```diff
@@ -53,14 +62,13 @@ exports.handler = async ({ source: { id: colonyAddress } }) => {
...
-        id: `${chainId}_${constants.AddressZero}_0`,
-        createdAt: now,
-        updatedAt: now,
-        createdAtBlock: block,
       };
     }
   }
-  return null;
+
+  // If the unclaimed balance is 0, or unclaimed balance is 0, still return a claim with amount zero.
```
[the remark this answers is not in the document]
Comment (davecreaser) [reply amplify/backend/function/fetchColonyNativeFundsClaim/src/index.js:71]: 🤣 busted.

**Comment by davecreaser** [comment]
@iamsamgibbs Thanks for the review! As to the behaviour you mentioned, yeah it's expected - good that you checked though. I've fixed the mobile icon shrinking and also the terribly worded comment 😆 Ready for re-review!

**Review by bassgeta — APPROVED** [review]
Nice and clean code changes!
Can confirm that ETH shows up
image
Sent over 5 ETH and it shows up:
image
And ETH then doesn't disappear:
image

Nice job pulling that row into a function instead of inlining it :ok_hand:

**Review by rdig — APPROVED** [review]
Seems to be working as advertised. Nice compromise on dealing with the chain native token!

Screenshot from 2024-08-26 16-41-24
Screenshot from 2024-08-26 16-43-07
Screenshot from 2024-08-26 16-43-17
Screenshot from 2024-08-26 16-43-45
Screenshot from 2024-08-26 16-44-22
Screenshot from 2024-08-26 16-44-43
Screenshot from 2024-08-26 16-45-05
Screenshot from 2024-08-26 16-45-56

**Review by iamsamgibbs — APPROVED** [review]
Nice one! Comment makes sense and pesky shrinking icon is resolved:

Screenshot 2024-08-27 at 11 39 03

Good to go!

**Pull request merged**